spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java

```diff
@@ -882,4 +892,48 @@ public String toString() {
 
 	}
 
+	/**
+	 * A strategy interface for conditional exception logging.
+	 * @since 1.5
+	 *
+	 */
+	public interface ConditionalExceptionLogger {
+
+		void log(Log logger, Throwable t);
+
+	}
+
+	/**
+	 * Default implementation of {@link ConditionalExceptionLogger} for logging channel
+	 * close exceptions.
+	 * @since 1.5
+	 *
+	 */
+	private class DefaultChannelCloseLogger implements ConditionalExceptionLogger {
```

Comment:
Looks like both these new classes are candidates to the top level - `org.springframework.amqp.utils`.
Otherwise this class must `static`.